src/common/models/ipmodels/ipblockputtag.go

```diff
@@ -0,0 +1,55 @@
+package ipmodels
+
+import (
+	ipapisdk "github.com/phoenixnap/go-sdk-bmc/ipapi"
+	files "phoenixnap.com/pnapctl/common/fileprocessor"
+)
+
+type TagAssignmentRequest struct {
+	Name  string  `yaml:"name" json:"name"`
+	Value *string `yaml:"value,omitempty" json:"value,omitempty"`
+}
+
+func PutIpBlockTagRequestFromFile(filename string, commandname string) ([]ipapisdk.TagAssignmentRequest, error) {
+	files.ExpandPath(&filename)
+
+	data, err := files.ReadFile(filename, commandname)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Marshal file into JSON using the struct
+	var TagAssignmentRequests []TagAssignmentRequest
+	err = files.Unmarshal(data, &TagAssignmentRequests, commandname)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return mapTagAssignmentRequestToSdk(&TagAssignmentRequests), nil
+}
+
+func (tagAssignmentRequest TagAssignmentRequest) ToSdk() ipapisdk.TagAssignmentRequest {
+	var tagAssignmentRequestSdk = ipapisdk.TagAssignmentRequest{
+		Name:  tagAssignmentRequest.Name,
+		Value: tagAssignmentRequest.Value,
+	}
+
+	return tagAssignmentRequestSdk
+}
+
+func mapTagAssignmentRequestToSdk(TagAssignmentRequest *[]TagAssignmentRequest) []ipapisdk.TagAssignmentRequest {
```

Comment:
I would update the naming for the function and parameter names to reflect that we're passing a list here. Also, the pointer should be removed here - so `[]TagAssignmentRequest`.

Reply:
Switched the param and name to TagAssignmentRequestList and removed pointer